### PR TITLE
feat(APIApplication): app authorization links and tags

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

**Reference Discord API Docs PRs or commits:**
- https://github.com/discord/discord-api-docs/pull/4112

### CODE REVIEW

File: deno/payloads/v8/application.ts [hunk 1]
```diff
@@ -91,6 +91,23 @@ export interface APIApplication {
 	 * See https://discord.com/developers/docs/resources/application#application-object-application-flags
 	 */
 	flags: ApplicationFlags;
+	/**
+	 * Up to 5 tags describing the content and functionality of the application
+	 */
+	tags?: [string, string, string, string, string];
```
Comment (ImRodry) [inline deno/payloads/v8/application.ts:97]: Only 1 tag is required if specified
```suggestion
	tags?: [string, string?, string?, string?, string?];
```

File: deno/payloads/v8/application.ts [hunk 2]
```diff
@@ -91,6 +91,23 @@ export interface APIApplication {
...
+	 */
+	install_params?: APIApplicationInstallParams;
+	/**
+	 * The application's default custom authorization link, if enabled
+	 */
+	custom_install_url?: string;
+}
+
+export interface APIApplicationInstallParams {
+	scopes: string[];
```
Comment (almeidx) [inline deno/payloads/v8/application.ts:109]: Could this type be used instead?
https://github.com/discordjs/discord-api-types/blob/fa54b9de5522b9fa9d5367650950f8b0e44f6e14/payloads/v9/oauth2.ts#L5

File: payloads/v9/application.ts [hunk 3]
```diff
@@ -91,6 +92,23 @@ export interface APIApplication {
...
+	install_params?: APIApplicationInstallParams;
+	/**
+	 * The application's default custom authorization link, if enabled
+	 */
+	custom_install_url?: string;
+}
+
+export interface APIApplicationInstallParams {
+	scopes: OAuth2Scopes[];
+	permissions: string;
```
Comment (vladfrangu) [inline payloads/v9/application.ts:111]: Use the Permissions type

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: payloads/v8/application.ts [hunk 4]
```diff
@@ -2,7 +2,8 @@
  * Types extracted from https://discord.com/developers/docs/resources/application
  */
 
-import type { Snowflake } from '../../globals';
+import type { OAuth2Scopes } from './index';
```
Comment (vladfrangu) [inline payloads/v8/application.ts:5]: can you import it from its actual file instead, or consolidate all other imports from this one?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by vladfrangu** [comment]
This needs a rebase

**Comment by popeda97** [comment]
suneettipirneni:feat/app-auth

**Pull request merged**